### PR TITLE
Fix #1045, Implement optimizer for is_scalar

### LINKED ISSUE #1045: Implement `is_scalar` function optimizer
http://php.net/manual/ru/function.is-scalar.php

```
bool is_scalar ( mixed $var )
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25420166-implement-is_scalar-function-optimizer?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F280146&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ovr**

### CODE REVIEW

File: unit-tests/Extension/Optimizers/IsScalarTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        $this->assertTrue($t->testVarParameter(1.0));
+        $this->assertTrue($t->testVarParameter(true));
+        $this->assertTrue($t->testVarParameter(false));
+        $this->assertTrue($t->testVarParameter(""));
+        $this->assertTrue($t->testVarParameter("test string"));
+
+        $this->assertFalse($t->testVarParameter([]));
+        $this->assertFalse($t->testVarParameter([1,2,3,4,5]));
+        $this->assertFalse($t->testVarParameter($this));
+        $this->assertFalse($t->testVarParameter(function() {}));
```
Comment (sergeyklay) [inline unit-tests/Extension/Optimizers/IsScalarTest.php]: What about null ? :)

Comment (ovr) [inline unit-tests/Extension/Optimizers/IsScalarTest.php]: @sergeyklay 
Just give me 1s

**Comment by ovr** [comment]
ping @andresgutierrez

**Comment by ovr** [comment]
Thanks @andresgutierrez

**Pull request merged**